src/server/pure.ts

```diff
@@ -13,7 +13,7 @@ function createServerRenderer<TProps, TResult>(
 ) {
   let renderProps: TProps | undefined
   let container: HTMLDivElement | undefined
-  let serverOutput: string = ''
+  let serverOutput = ''
   const testHarness = createTestHarness(rendererProps, wrapper, false)
 
   return {
```

Comment:
github won't let me comment on the actual line, but the error in the test is being caused by the `act` call on line 22.  For some reason, when you use `react-dom@16.9.0` this tried to use the document which throws an error when testing in the node environment, which doesn't occur in newer versions.

You can replicate this by running:

```sh
npm run install:react-16.9.0
npm test -- --no-watch src/__tests__/ssr.test.ts
```

In the context of server rendering, the `act` call is not really necessary here.  Dropping it out still passes all of our tests and allows your new test to pass with the older version:

```ts
    render(props?: TProps) {
      renderProps = props
      try {
        serverOutput = ReactDOMServer.renderToString(testHarness(props))
      } catch (e: unknown) {
        rendererProps.setError(e as Error)
      }
    },
```

Reply:
Wow, I don't think I would have figured that out on my own 😅 I'll test this out and make the necessary changes. Thanks a bunch!